integration/src/main/java/scripts/testscripts/WorkspacePermissionOperations.java

```diff
@@ -0,0 +1,173 @@
+    testReaderPermissions();
+    testNoPermissions();
+    testAdminPermissionsOnUserSnapshot();
+  }
+
+  private void testReaderPermissions() throws Exception {
+    setTestWorkspacePermissionForRegularUser(WorkspaceAccessLevel.READER.getValue());
+    // User cannot create a catalog entry on snapshot when user only has reader access on snapshot
+    CreateDatasetRequest request = datasetRequestForWorkspace(adminTestWorkspace.getWorkspaceId());
+    // note if this assertion fails we'll leave behind a stray dataset in the db
+    assertThrows(ApiException.class, () -> userDatasetsApi.createDataset(request));
+    assertThat(
+        userDatasetsApi.getApiClient().getStatusCode(),
+        is(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED));
+
+    // verify the user can still access data though
+    userDatasetsApi.getDataset(adminTestDatasetId);
+    assertThat(userDatasetsApi.getApiClient().getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
+  }
+
+  private void testNoPermissions() throws Exception {
+    clearTestWorkspacePermissions();
+    // verify the user cannot access dataset without any permissions
+    assertThrows(ApiException.class, () -> userDatasetsApi.getDataset(adminTestDatasetId));
+    assertThat(
+        userDatasetsApi.getApiClient().getStatusCode(), is(HttpStatusCodes.STATUS_CODE_NOT_FOUND));
+  }
+  // TODO (DC-446): Fix with https://broadworkbench.atlassian.net/browse/DC-446
+  //  private void testNoAccessPermissions() throws Exception {
+  //    setTestWorkspacePermissionForRegularUser(WorkspaceAccessLevel.NO_ACCESS.getValue());
+  //    // User cannot create a catalog entry on Workspace when user has "No Access" on a workspace
+  //    CreateDatasetRequest request =
+  // datasetRequestForWorkspace(adminTestWorkspace.getWorkspaceId());
+  //    // note if this assertion fails we'll leave behind a stray dataset in the db
+  //    assertThrows(ApiException.class, () -> userDatasetsApi.createDataset(request));
+  //    assertThat(
+  //        userDatasetsApi.getApiClient().getStatusCode(),
+  // is(HttpStatusCodes.STATUS_CODE_NOT_FOUND));
+  //    // but the user can get datasets
+  //    userDatasetsApi.getDataset(adminTestDatasetId);
+  //    assertThat(userDatasetsApi.getApiClient().getStatusCode(),
+  // is(HttpStatusCodes.STATUS_CODE_OK));
+  //  }
+
+  private void testAdminPermissionsOnUserSnapshot() throws Exception {
+    // Verify admin can create a dataset on the user snapshot
+    CreateDatasetRequest request = datasetRequestForWorkspace(userTestWorkspace.getWorkspaceId());
+    adminCreateDataset(request);
+  }
+
+  private void setTestWorkspacePermissionForRegularUser(String policy) throws Exception {
+    // first clear the shared snapshot of policy state caused by previous tests
+    adminRawlsClient.updateWorkspaceAcl(
+        List.of(new WorkspaceACLUpdate().email(regularUser.userEmail).accessLevel(policy)),
+        adminTestWorkspace);
+  }
```

Comment:
If `adminRawlsClient.updateWorkspaceAcl()` supported an API like `adminRawlsClient.updateWorkspaceAcl(adminTestWorkspace, regularUser.userEmail, WorkspaceAccessLevel.READER)` then maybe this method wouldn't be needed anymore, since most of the mess of making the ACL list would be hidden from this code.